test/test_timestamp.py

```diff
@@ -86,6 +86,24 @@ def test_timestamp_datetime():
     utc = datetime.timezone.utc
     assert t.to_datetime() == datetime.datetime(1970, 1, 1, 0, 0, 42, 0, tzinfo=utc)
 
+    ts = datetime.datetime(2024, 4, 16, 8, 43, 9, 420317, tzinfo=utc)
+    ts2 = datetime.datetime(2024, 4, 16, 8, 43, 9, 420318, tzinfo=utc)
+
+    assert Timestamp.from_datetime(ts2).nanoseconds - Timestamp.from_datetime(ts).nanoseconds == 1e3
+
+    ts3 = datetime.datetime(2024, 4, 16, 8, 43, 9, 4256)
+    ts4 = datetime.datetime(2024, 4, 16, 8, 43, 9, 4257)
+    assert (
+        Timestamp.from_datetime(ts4).nanoseconds - Timestamp.from_datetime(ts3).nanoseconds == 1e3
+    )
+
+    assert Timestamp.from_datetime(ts).to_datetime() == ts
+
+    t2 = Timestamp(1713256989, 420318123)
+    t3 = Timestamp(1713256989, 420318499)
+    t4 = Timestamp(1713256989, 420318501)
+    assert t2.to_datetime() == t3.to_datetime() != t4.to_datetime()
```

Comment:
Maybe just using 2 asserts would be simpler.

Reply:
definitely, we dont necessarily need to compare with t2 now i think of it. But these were to test the rounding specifically, if rounding is wrong way to go, these tests arent necessarily needed imo.